src/OrleansServiceBus/Providers/Streams/EventHub/CachePressureMonitors/SlowConsumingPressureMonitor.cs

```diff
@@ -0,0 +1,104 @@
+        private const double DefaultFlowControlThreshold = 0.5;
+
+        /// <summary>
+        /// PressureWindowSize
+        /// </summary>
+        public TimeSpan PressureWindowSize { get; set; }
+        /// <summary>
+        /// FlowControlThreshold
+        /// </summary>
+        public double FlowControlThreshold { get; set; }
+
+        private readonly Logger logger;
+        private double biggestPressureInCurrentPeriod;
+        private DateTime nextCheckedTime;
+        private bool isUnderPressure;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(Logger logger)
+            : this(DefaultFlowControlThreshold, DefaultPressureWindowSize, logger)
+        { }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="pressureWindowSize"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(TimeSpan pressureWindowSize, Logger logger)
+            : this(DefaultFlowControlThreshold, pressureWindowSize, logger)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, Logger logger)
+            : this(flowControlThreshold, DefaultPressureWindowSize, logger)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="pressureWindowSzie"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, TimeSpan pressureWindowSzie, Logger logger)
+        {
+            this.FlowControlThreshold = flowControlThreshold;
+            this.logger = logger.GetSubLogger(this.GetType().Name);
+            this.nextCheckedTime = DateTime.MinValue;
+            this.biggestPressureInCurrentPeriod = 0;
+            this.isUnderPressure = false;
+            this.PressureWindowSize = pressureWindowSzie;
+        }
+
+        public void RecordCachePressureContribution(double cachePressureContribution)
+        {
+            if (cachePressureContribution > biggestPressureInCurrentPeriod)
+                biggestPressureInCurrentPeriod = cachePressureContribution;
+        }
+
+        public bool IsUnderPressure(DateTime utcNow)
+        {
+            //if any pressure contribution in current period is bigger than flowControlThreshold
+            //we see the cache is under pressure
+            bool underPressure = this.biggestPressureInCurrentPeriod > this.FlowControlThreshold;
+            if (this.isUnderPressure != underPressure)
+            {
+                this.isUnderPressure = underPressure;
+                logger.Verbose(this.isUnderPressure
```

Comment:
Sure, on a state change makes more sense. But still it would be nice to know the details for when an individual monitor starts throttling. Maybe in the individual monitor, log as Info only when it transitions to under pressure when it wasn't (but not necessarily log as Info when it goes back to not under pressure, at that point the aggregated monitor logger can be enough).